CONTRIBUTING.md

```diff
@@ -31,36 +31,151 @@ Most of nixvim is dedicated to wrapping neovim plugins such that we can configur
-     e.g. `"foo-bar-nvim` for `pkgs.vimPlugins.foo-bar-nvim`, or `[ "hello" "world" ]` for `pkgs.hello.world`.
-  - `maintainers`: Register yourself as a maintainer for this plugin:
-    - e.g. `maintainers = [ lib.maintainers.JosephFourier ]`
-    - If you are already registered as a [`nixpkgs` maintainer][nixpkgs-maintainers], there is nothing more to do.
-    - Otherwise, you should add yourself to our [`maintainers.nix`](lib/maintainers.nix) list.
-    - See the documentation at the top of the [`nixpkgs` maintainers list][nixpkgs-maintainers] for more detail.
-  - `settingsOptions`: All or some (only the most important ones) option declarations for this plugin settings.\
-    See below for more information
-  - `settingsExample`: An example of what could the `settings` attrs look like.
-
-[nixpkgs-maintainers]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix
+3. Add the necessary arguments when calling either [`mkNeovimPlugin`](#mkneovimplugin) or [`mkVimPlugin`](#mkvimplugin)
+
+#### `mkNeovimPlugin`
+
+The `mkNeovimPlugin` function provides a standardize way to create a `Neovim` plugin.
+This is intended to be used with lua plugins that have a `setup` function,
+although it is flexible enough to be used with similar variants of plugins.
+A template plugin can be found in (plugins/TEMPLATE.nix)[https://github.com/nix-community/nixvim/blob/main/plugins/TEMPLATE.nix].
+
+| Parameter                    | Description                                                                                                                                                                                                                                          | Required | Default Value                                                                                   |
+| ---------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------- | ----------------------------------------------------------------------------------------------- |
+| **name**                     | The name of the plugin.                                                                                                                                                                                                                              | Yes      | N/A                                                                                             |
+| **url**                      | The URL of the plugin's repository.                                                                                                                                                                                                                  | Yes      | N/A                                                                                             |
```

Comment:
https://github.com/nix-community/nixvim/pull/2462#discussion_r1818741351 is still outstanding. `url` defaults to "default package's `meta.homepage` if available".

```suggestion
| **url**                      | The URL of the plugin's repository.                                                                                                                                                                                                                  | Yes      | `package` parameter's `meta.homepage`                                                           |
```

Reply:
Yup, forgot the modification when to void of my mind. Corrected